src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+            mHm_hs_c *home = &Save_Get(homes)[mHS_get_arrange_idx(Common_Get(player_no))];
+            int action;
+            int choice = mChoice_Get_ChoseNum(mChoice_Get_base_window_p());
+            switch (choice) {
+            case 0:
+            case 1:
+            case 2:
+                action = 0x2;
+                home->size_info.next_size += 1;
+                home->ordered_outlook_pal = choice;
+                aNSC_set_rehouse_order_date(home);
+                break;
+            default:
+                action = 0x5;
+                break;
+            }
+            mDemo_Set_OrderValue(mDemo_ORDER_NPC0, 9, 0x0);
+            aNSC_setupAction(shop_master, game, action);
+        }
+    }
+}
+
+static void aNSC_check_roof_col_order2(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    if (mMsg_Check_MainNormalContinue(mMsg_Get_base_window_p()) == TRUE) {
+        mHm_hs_c *home = &Save_Get(homes)[mHS_get_arrange_idx(Common_Get(player_no))];
+        int choice = mChoice_Get_ChoseNum(mChoice_Get_base_window_p());
+        int unk = mDemo_Get_OrderValue(mDemo_ORDER_NPC0, 9);
+        int action = 0;
+        switch (choice) {
+        case 0:
+        case 1:
+        case 2:
+            action = 0x2;
+            home->size_info.next_size += 1;
+            home->ordered_outlook_pal = choice + unk * 3;
+            aNSC_set_rehouse_order_date(home);
+            break;
+        default:
+            action = 0x5;
+        }
+        mDemo_Set_OrderValue(mDemo_ORDER_NPC0, 9, 0x0);
+        aNSC_setupAction(shop_master, game, action);
+    }
+}
+
+static void aNSC_present_balloon_start_wait(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    if ((int)mDemo_Get_OrderValue(0x4, 0x1) == 2) {
+        aNSC_setupAction(shop_master, game, 0x8);
+    }
+}
+
+static void aNSC_present_balloon_trans_move(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    ACTOR *actorx = (ACTOR *)shop_master;
+    if (actorx->world.position.z > (f32)260.0) {
+        actorx->world.position.z = (f32)260.0;
+        aNSC_setupAction(shop_master, game, 0x9);
+    }
+}
+
+static void aNSC_present_balloon_trans_takeout(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    ACTOR *actorx = (ACTOR *)shop_master;
+    ACTOR *res;
+    if (shop_master->npc_class.left_hand.item_actor_p == NULL) {
+        res = CLIP(handOverItem_clip)->birth_proc(ITM_BALLOON_START, aHOI_REQUEST_PUTAWAY, aHOI_PRESENT, (ACTOR *)shop_master);
+        if (res) {
+            CLIP(handOverItem_clip)->chg_request_mode_proc((ACTOR *)shop_master, aHOI_REQUEST_TRANSFER);
+
+            shop_master->npc_class.left_hand.item_actor_p = res;
+        }
+    }
+
+    if (shop_master->npc_class.draw.main_animation_state == cKF_STATE_STOPPED) {
+        if (CLIP(handOverItem_clip)->master_actor == (ACTOR *)shop_master) {
+            CLIP(handOverItem_clip)->chg_request_mode_proc((ACTOR *)shop_master, 0x2); 
```

Comment:
We have enums for the hand over item system. They'd be using `aHOI_REQUEST_*`. This one is `aHOI_REQUEST_TRANS_WAIT`.